src/Helper/HeaderHelper.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace Yiisoft\Yii\Web\Helper;
+
+use Psr\Http\Message\RequestInterface;
+
+final class HeaderHelper
+{
+    /**
+     * Explode header value to value and parameters (eg. text/html;q=2;version=6)
+     * @param string $headerValue
+     * @return array first element is the value, and key-value are the parameters
+     */
+    public static function getValueAndParameters(string $headerValue): array
+    {
+        $headerValue = trim($headerValue);
+        if (strlen($headerValue) === 0) {
+            return [];
+        }
+        $parts = preg_split('/\s*;\s*/', $headerValue, -1, PREG_SPLIT_NO_EMPTY);
+        $output = [array_shift($parts)];
+        foreach ($parts as $part) {
+            [$key, $headerValue] = explode('=', $part, 2);
+            $output[$key] = $headerValue;
+        }
+        return $output;
+    }
+
+    /**
+     * Getting header value as q factor sorted list
+     * @param string|string[] $values Header value as a comma-separated string or already exploded string array.
+     * @see getValueAndParameters
+     * @link https://developer.mozilla.org/en-US/docs/Glossary/Quality_values
+     */
+    public static function getByQFactorSortedList($values, bool $outputValuesOnly = false): array
```

Comment:
```suggestion
    public static function getSortedValueAndParameters($values, bool $outputValuesOnly = false): array
```

Reply:
Name is longer but, I think, is a bit more readable.

Reply:
It's not clear to me what is expected as input value for `$values`.

Reply:
Also I don't like `$outputValuesOnly`. It looks like method is doing too much and could be actually split into two more specific and simpler methods (or more).

Reply:
> Also I don't like `$outputValuesOnly`. It looks like method is doing too much and could be actually split into two more specific and simpler methods (or more).

Really, this parameter is not required. Later I saw that I can't use it in to accept the header, because the process itself is so complicated. In my opinion, post-processing is generally required, so this parameter is not relevant.

Reply:
> Name is longer but, I think, is a bit more readable.

I check to see if there is any header content that is sorted differently, because in that case the method name might be problematic.